### PR TITLE
implemented some lobby search options

### PR DESCRIPTION
**Pull request by daki4**
I have implemented some of the lobby functions (distance filter, string filter, numerical filter, and also request_lobby_data).

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -543,6 +543,8 @@ impl GameId {
 
 #[cfg(test)]
 mod tests {
+    use std::io::{Read, Write};
```
Comment (ceifa) [inline src/lib.rs:546]: Why this is needed?

Comment (daki4) [reply src/lib.rs:546]: That was not meant to be included.... That was a part of a test, but I have forgotten to take it out, I will remove it, and resubmit.

File: src/matchmaking.rs [hunk 2]
```diff
@@ -71,6 +91,86 @@ impl<Manager> Matchmaking<Manager> {
         }
     }
 
+    /// set lobby distance search filter
+    ///
+    pub fn add_lobby_distance_filter(&self, filter: LobbyDistanceFilter)
+    {
+        unsafe {
+            sys::SteamAPI_ISteamMatchmaking_AddRequestLobbyListDistanceFilter(
+                sys::SteamAPI_SteamMatchmaking_v009(),
```
Comment (ceifa) [inline src/matchmaking.rs:100]: Use `self.mm` instead of calling `SteamAPI_SteamMatchmaking_v009` every time

**Comment by ceifa** [comment]
Implemented in #141

**Pull request closed without merging**